test/functional/rpc_budget.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 The PIVX developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test RPC commands for budget proposal creation, submission, and verification."""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+
+
+class BudgetProposalTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def run_test(self):
+        budgetcycleblocks = 144
+        nextsuperblock = self.nodes[0].getnextsuperblock()
+        address = self.nodes[0].getnewaddress()
+        scheme = 'http://'
+        url = 'test.com'
+        longurl = 'test.com/longer-than-allowed-url-string-should-fail-if-url-is-more-than-64-characters'
+        name = 'testinvalid'
+        longname = 'ThisProposalNameHasMoreThanTwentyCharacters'
```

Comment:
Test parameters should be exactly on the borders of the testing size.  e.g.
```
url = <57 character URL>
longurl = <58 character URL>
name = <20 character name>
longname = <21 character name>
```
Rather than having plenty of wiggle room in the test case; as a change from `<` to `<=` wouldn't be caught.  (And ensuring the max limit actually will pass the code)

Reply:
done